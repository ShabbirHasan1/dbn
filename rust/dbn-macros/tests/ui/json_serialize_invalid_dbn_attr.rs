use dbn_macros::JsonSerialize;

#[derive(JsonSerialize)]
#[repr(C)]
struct Record {
    #[dbn(unknown)]
    pub a: u8,
}

fn main() {}
