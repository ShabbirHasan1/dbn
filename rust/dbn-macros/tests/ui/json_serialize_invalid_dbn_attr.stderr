error: Invalid attr `unknown` passed to `#[dbn]`
 --> tests/ui/json_serialize_invalid_dbn_attr.rs:6:11
  |
6 |     #[dbn(unknown)]
  |           ^^^^^^^

error[E0433]: failed to resolve: could not find `encode` in the crate root
 --> tests/ui/json_serialize_invalid_dbn_attr.rs:3:10
  |
3 | #[derive(JsonSerialize)]
  |          ^^^^^^^^^^^^^ could not find `encode` in the crate root
  |
  = note: this error originates in the derive macro `JsonSerialize` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0433]: failed to resolve: could not find `json_writer` in the crate root
 --> tests/ui/json_serialize_invalid_dbn_attr.rs:3:10
  |
3 | #[derive(JsonSerialize)]
  |          ^^^^^^^^^^^^^ could not find `json_writer` in the crate root
  |
  = note: this error originates in the derive macro `JsonSerialize` (in Nightly builds, run with -Z macro-backtrace for more info)
